mass_mailing_custom_unsubscribe/tests/test_ui.py

```diff
@@ -0,0 +1,176 @@
+# Copyright 2016 Jairo Llopis <jairo.llopis@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+import mock
+from contextlib import contextmanager
+from odoo.tests.common import HttpCase
+from werkzeug import urls
+
+
+class UICase(HttpCase):
+    _tour_run = "odoo.__DEBUG__.services['web_tour.tour'].run('%s')"
+    _tour_ready = "odoo.__DEBUG__.services['web_tour.tour'].tours.%s.ready"
+
+    def extract_url(self, mail, *args, **kwargs):
+        url = mail._get_unsubscribe_url(self.email)
+        self.assertTrue(urls.url_parse(url).decode_query().get('token'))
+        self.assertTrue(url.startswith(self.domain))
+        self.url = url.replace(self.domain, "", 1)
+        return True
+
+    def setUp(self):
+        super(UICase, self).setUp()
+        self.email = "test.contact@example.com"
+        self.mail_postprocess_patch = mock.patch(
+            "odoo.addons.mass_mailing.models.mail_mail.MailMail."
+            "_postprocess_sent_message",
+            autospec=True,
+            side_effect=self.extract_url,
+        )
+        with self.tempenv() as env:
```

Comment:
Remove this `with` block and dedent inner lines, and make them use `self.env` instead of `env`.

This is a workaround for v11 and lower versions, but in v12 it's not needed nor helpful.